lib/ansible/executor/task_queue_manager.py

```diff
@@ -195,7 +197,12 @@ def load_callbacks(self):
                 raise AnsibleError("Invalid callback for stdout specified: %s" % self._stdout_callback)
             else:
                 self._stdout_callback = callback_loader.get(self._stdout_callback)
-                self._stdout_callback.set_options()
+                callback_variables = {}
+                for var_name in C.config.get_plugin_vars('callback', self._stdout_callback._load_name):
+                    if var_name not in self._variable_manager._extra_vars:
+                        continue
+                    callback_variables[var_name] = templar.template(self._variable_manager._extra_vars[var_name])
+                self._stdout_callback.set_options(var_options=callback_variables)
```

Comment:
This block of code appears to be nearly identical to the one below. Can you extract it into a function?

Also, the variable lookup from `_extra_vars` should be able to use a walrus operator to simplify the logic, using a sentinel value that won't appear in the data. Something like this:

```python
if value := self._variable_manager._extra_vars.get(var_name, ...) is not ...:
    callback_variables[var_name] = templar.template(value)
```

Reply:
I moved this into a callback method (since the connection plugins also have one), and simplified it as you suggested.

Reply:
i would keep out of the callback itself to avoid passing variables to the plugin that it should not have access to. Right now callbacks execute in the same process, but if we ever want to isolate it, this would disclose more info that would be advisable.

Reply:
@bcoca is this something we should change for connection plugins?